<?xml version="1.0" encoding="utf-8"?>

<launch>
    <arg name="sphinx" default="true"/>
    <arg name="unity" default="true"/>
    <arg name="physical" default="false"/>
    <arg name="vicon" default="false"/>
    <arg name="multiview" default="false"/>

    <include file="$(find drone_controller)/launch/main.launch">
        <arg name="launch_unity" value="$(arg unity)" />
        <arg name="launch_shinx" value="$(arg sphinx)" />
        <arg name="launch_physical" value="$(arg physical)" />
        <arg name="launch_vicon" value="$(arg vicon)" />
        <arg name="launch_multiview" value="$(arg multiview)" />
    </include>

    <node pkg="drone_controller" name="program_controller_node" type="gate_scenario_controller.py" output="screen">
    </node>

    <!-- Error is a value between -1 an 1, and the output is a value between -100, 100 -->
    <node pkg="drone_controller" name="gate_navigation_node" type="gate_navigation.py" output="screen">
        <param name="gains/p" type="double" value="27" />
        <param name="gains/i" type="double" value="0" />
        <param name="gains/d" type="double" value="15" />
    </node> 

    <node pkg="mixing_functions" name="color_threshold_node" type="color_threshold.py" output="screen">
        <param name="lower_bound/H" type="int" value="0" />
        <param name="lower_bound/S" type="int" value="0" />
        <param name="lower_bound/V" type="int" value="40" />
        <param name="upper_bound/H" type="int" value="20" />
        <param name="upper_bound/S" type="int" value="200" />
        <param name="upper_bound/V" type="int" value="200" />

        <param name="input_topic" type="string" value="/uav1/sphinx_sensors/camera" />
        <param name="output_topic" type="string" value="/uav1/overlay/camera" />
    </node> 

    <group if="$(arg multiview)">
        <node pkg="rqt_image_view" type="rqt_image_view" name="gate_center" args="/gate_navigation/sensors/camera" output="screen"/>
        <node pkg="rqt_image_view" type="rqt_image_view" name="test_color" args="/uav1/overlay/camera" output="screen"/>
    </group>

</launch>